mirgecom/gas_model.py

```diff
@@ -350,9 +354,7 @@ def make_fluid_state(cv, gas_model,
     smoothness_beta = (actx.np.zeros_like(cv.mass) if smoothness_beta
                        is None else smoothness_beta)
 
-    # the porous media is identified if the wall density is not None
-
-    if material_densities is None:
+    if isinstance(gas_model, GasModel):
```

Comment:
It can be the `PorousFlowModel`. This is one of the main differences from the previous way where wall and gas were mixed together.

Reply:
The point I was trying to make is that in principle if it's called `gas_model` and documented as a `GasModel`, it should be something that derives from `GasModel`. I'm OK with weakening that if it's an intermediate step towards #930, but I would add a FIXME here explaining what's going on in order to make it less alarming.

Reply:
It is an intermediate step. I would rename everything from `gas_model` to something like `flow_model`. Since everything in the code calls `make_fluid_state`, I am avoiding any major changes/refactor now.

I will add a FIXME/TODO to clarify this point and as a future reminder.